caffe2/core/c10_operator.h

```diff
@@ -26,37 +37,39 @@ inline void _call_caffe2_op(const c10::FunctionSchema& schema, std::vector<c10::
 inline void _call_caffe2_op_from_c10(
     c10::Stack* stack,
     const c10::FunctionSchema& schema,
-    at::DeviceType deviceType,
     _CallCaffe2OpFunc* call_op) {
-  // precondition: on the stack, there's an IValue for each caffe2 input and an IValue for each caffe2 output.
-  // (note: in the jit schema, these caffe2 outputs are explicitly listed as additional inputs).
-  // The output ones could either be a preallocated tensor or ivalue::None.
-
-  const size_t num_outputs = schema.returns().size();
-  const size_t total_num_arguments = schema.arguments().size();
-  const size_t num_inputs = total_num_arguments - num_outputs;
+  // precondition: on the stack, there's one IValue for each argument of the
+  // c10 schema. The last argument is an optional tensor list that
+  // (if not ivalue::None) contains a preallocated output tensor for each
+  // operator output.
 
-  // TODO Avoid vector allocation. One idea would be to keep the std::vector instances in the cache.
-  auto outputs = torch::jit::pop(*stack, num_outputs);
-  auto inputs = torch::jit::pop(*stack, num_inputs);
+  AT_ASSERT(schema.arguments().back().type()->isSubtypeOf(
```

Comment:
Another assumption that the operator has more than zero elements

Reply:
This is correct here since when a caffe2 operator is wrapped into c10, it always adds the last parameter for preallocated outputs. I should assert that the number of arguments is larger than zero though before blindly accessing back.